libraries/AP_Relay/AP_Relay.cpp

```diff
@@ -152,6 +152,18 @@ void AP_Relay::toggle(uint8_t instance)
     }
 }
 
+// check settings are valid
+bool AP_Relay::arming_checks(size_t buflen, char *buffer) const
+{
+    for (uint8_t i=0; i<AP_RELAY_NUM_RELAYS; i++) {
+        if (_pin[i] != -1 && !hal.gpio->valid_pin(_pin[i])) {
+            hal.util->snprintf(buffer, buflen, "Relay[%u] pin %d invalid", i + 1, _pin[i]);
```

Comment:
I'm surprised you don't have to cast the "i+1" to (unsigned) to avoid a compile error.  Perhaps that has gone away with a compiler upgrade.

Reply:
it seems happy